AU2/plugins/custom_plugins/PolicePlugin.py

```diff
@@ -0,0 +1,228 @@
+  {ROWS}
+</table>
+"""
+
+POLICE_TABLE_ROW_TEMPLATE = """
+<tr><td>{RANK}</td><td>{PSEUDONYM}</td><td>{NAME}</td><td>{EMAIL}</td><td>{COLLEGE}</td><td>{NOTES}</td></tr>
+"""
+
+DEAD_POLICE_TABLE_TEMPLATE = """
+<p xmlns="">
+      Killed in the line of duty:
+   </p>
+<table xmlns="" class="playerlist">
+  <tr><th>Rank</th><th>Pseudonym</th><th>Real Name</th><th>Email Address</th><th>College</th><th>Notes</th></tr>
+  {ROWS}
+</table>
+"""
+
+DEAD_POLICE_TABLE_ROW_TEMPLATE = """
+<tr><td>{RANK}</td><td>{PSEUDONYM}</td><td>{NAME}</td><td>{EMAIL}</td><td>{COLLEGE}</td><td>{NOTES}</td></tr>
+"""
+
+DEAD_CORRUPT_POLICE_TABLE_TEMPLATE = """
+<p xmlns="">
+      And these are the names of those who turned to the dark side, and paid the price:
+   </p>
+<table xmlns="" class="playerlist">
+  <tr><th>Rank</th><th>Pseudonym</th><th>Real Name</th><th>Email Address</th><th>College</th><th>Notes</th></tr>
+  {ROWS}
+</table>
+"""
+
+DEAD_CORRUPT_POLICE_TABLE_ROW_TEMPLATE = """
+<tr><td>{RANK}</td><td>{PSEUDONYM}</td><td>{NAME}</td><td>{EMAIL}</td><td>{COLLEGE}</td><td>{NOTES}</td></tr>
+"""
+
+NO_DEAD_POLICE = """<p xmlns="">No police have been killed... yet.</p>"""
+
+POLICE_PAGE_TEMPLATE: str
+with open(os.path.join(ROOT_DIR, "plugins", "custom_plugins", "html_templates", "police.html"), "r") as F:
+    POLICE_PAGE_TEMPLATE = F.read()
+
+@registered_plugin
+class PolicePlugin(AbstractPlugin):
+    def __init__(self):
+        super().__init__("PolicePlugin")
+
+        self.html_ids = {
+            "Rank": self.identifier + "_rank",
+            "Rank Name": self.identifier + "_name"
+        }
+
+        self.config_exports = [
+            ConfigExport(
+                "PolicePlugin_set_ranks",
+                "Police -> Set Ranks",
+                self.ask_set_ranks,
+                self.answer_set_ranks
+            )
+        ]
+
+        self.ranks = GENERIC_STATE_DATABASE.arb_state.setdefault(self.identifier, DEFAULT_RANKS)
+
+    def ask_set_ranks(self):  # TODO Make Ctrl-C work - not sure why it isn't
+        return [
+            Label("Rename different ranks of police to fit the game theme"),
+            Label("Ranks affect position on the police page, and can be used to reward/punish/just have a bit of fun"),
+            InputWithDropDown(
+                title="Select rank to rename",
+                identifier=self.html_ids["Rank"],
+                options=[str(i[0]) + ': ' + i[1] for i in list(GENERIC_STATE_DATABASE.arb_state.get(self.identifier).items())]
+            ),
+            NamedSmallTextbox(
+                title="Enter new name",
+                identifier=self.html_ids["Rank Name"],
+            )
+        ]
+
+    def answer_set_ranks(self, htmlResponse):
+        rank = htmlResponse[self.html_ids['Rank']]
+        new_name = htmlResponse[self.html_ids['Rank Name']]
+        if new_name:
+            GENERIC_STATE_DATABASE.arb_state[self.identifier][rank[0]] = new_name
```

Comment:
You need to play evasively whenever setting into a dictionary (as you've done elsewhere), e.g.:

```suggestion
            GENERIC_STATE_DATABASE.arb_state.setdefault(self.identifier, {})[rank[0]] = new_name
```